src/sidebar/services/test/frame-sync-test.js

```diff
@@ -861,13 +861,89 @@ describe('FrameSyncService', () => {
   describe('#scrollToAnnotation', () => {
     beforeEach(async () => {
       frameSync.connect();
+    });
+
+    it('does nothing if matching guest frame is not found', async () => {
+      frameSync.scrollToAnnotation(fixtures.ann);
+    });
+
+    function createEPUBAnnotation(cfi) {
+      return {
+        id: 'epub-id-1',
+        $tag: 'epub-tag-1',
+        target: [
+          {
+            selector: [
+              {
+                type: 'EPUBContentSelector',
+                cfi,
+              },
+            ],
+          },
+        ],
+      };
+    }
+
+    it('should scroll to the annotation in the correct guest', async () => {
       await connectGuest();
+      emitGuestEvent('documentInfoChanged', fixtures.htmlDocumentInfo);
+
+      frameSync.scrollToAnnotation(fixtures.ann);
+
+      assert.calledWith(
+        guestRPC().call,
+        'scrollToAnnotation',
+        fixtures.ann.$tag
+      );
     });
 
-    it('should scroll to the annotation in the guest', () => {
-      frameSync.connect();
-      frameSync.scrollToAnnotation('atag');
-      assert.calledWith(guestRPC().call, 'scrollToAnnotation', 'atag');
+    it('should trigger a navigation in an EPUB if needed', async () => {
```

Comment:
This test is clearly structured and commented, but a bit...long. Not that I have a better suggestion.